06.bowling_object/game.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+require_relative 'shot'
+
+class Game
+  def initialize(marks)
+    shots = marks.map { |mark| Shot.new(mark) }
+
+    shots_index = 0
+    @frames = []
+    0.upto(9).each do |frame_number|
+      length =
+        if frame_number == 9
+          3
+        elsif shots[shots_index].strike?
+          1
+        else
+          2
+        end
+      @frames << Frame.new(frame_number, *shots[shots_index, length])
+      shots_index += length
+    end
```

Comment:
```suggestion
    @marks = marks
```

initializeであれこれロジックを書くと、newするだけで時間を食ったり、バグを踏んでエラーが出たりする可能性が出てきます。

initializeはなるべくシンプルにして、本来処理すべきメソッド（ここでは`calc_result`）にロジックを移行させると良いかもしれません。

もしくは、Gameはnewさせない（＝クラスメソッドだけにする）、というアプローチもありですね。

```ruby
marks = ARGV[0].chomp.split(',')
puts Game.calc_result(marks)
```